qt-bugs@ issue : none yet
Trolltech task ID : none yet
bugs.kde.org number : ?
applied: no
author: Lubos Lunak <l.lunak@kde.org>

Fix QDesktopWidget not detecting some changes in multiscreen setups
(e.g. two screens 1280x1024, one of them switching to smaller size).

--- src/gui/kernel/qapplication_x11.cpp.sav	2008-08-21 16:33:15.000000000 +0200
+++ src/gui/kernel/qapplication_x11.cpp	2008-08-27 12:43:06.000000000 +0200
@@ -3001,11 +3001,11 @@ int QApplication::x11ProcessEvent(XEvent
         QSize oldSize(w->size());
         w->data->crect.setWidth(DisplayWidth(X11->display, scr));
         w->data->crect.setHeight(DisplayHeight(X11->display, scr));
-        if (w->size() != oldSize) {
+//        if (w->size() != oldSize) {
             QResizeEvent e(w->size(), oldSize);
             QApplication::sendEvent(w, &e);
             emit desktop()->resized(scr);
-        }
+//        }
     }
 #endif // QT_NO_XRANDR
 
